### PR TITLE
harden export account

### PR DESCRIPTION
**Pull request by witmicko**
## Explanation

Fixes [security #113](https://github.com/MetaMask/pm-security/issues/113), by adding password check in the backend api.

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by pedronfigueiredo — DISMISSED** [review]

File: ui/store/actions.test.js [hunk 1]
```diff
@@ -1245,6 +1245,7 @@ describe('Actions', () => {
     });
   });
 
+  // eslint-disable-next-line jest/no-focused-tests
```
Comment (weizman) [inline ui/store/actions.test.js:1248]: why is this necessary?

**Review by weizman — DISMISSED** [review]
LGTM thank you for this important fix!

**Review by pedronfigueiredo — DISMISSED** [review]

**Review by pedronfigueiredo — APPROVED** [review]

**Review by NEllusion — APPROVED** [review]
